### PR TITLE
Logging base

### PR DESCRIPTION
**Pull request by sammeboy635**
I know this is a big PR but all of this is needed for the actions and logging to work together.  

Logging PR:

- New internal/wptutil.py: which adds logging using the existing logging lib.
- New  --debug argument that enables logs, prevents deletion of work files, bypasses idle check for more consistent performance results, and ran with (--testurl (light, medium, heavy)) will automatically set standard testing URL.
- Logs are now through out the code base
- Actions now runs local (Light, Heavy) test, Prints log timing, Verbose runs to each commit, and compares new functions vs old in devtools_parsing.py.
- vvvvv is a new logging format with colors, so we can see errors easily. It also creates logging/main.log which is great for seeing the code flow and timings throughout.

Please Check out the actions for this PR to see all the new tests we have.

### CODE REVIEW

**Comment by sammeboy635** [comment]
@pmeenan Feel free to let me know what you think!

**Review by pmeenan — CHANGES_REQUESTED** [review]
Sorry, quite a few comments but at a high level:
* This mixes logging, testing and some utility functions. Separate PR's for each would be a lot cleaner.
* A readme document the best-practices for when to use each logging level is needed.
* The additional log levels should be given useful names
* Would probably be cleaner as a custom logging module that overrides the default logger and adds the new levels (and takes care of setting up the filtering, etc)
* "wptutil" looks like it does a lot of different things, some of which should be moved and the remainder probably require a different module name and should probably be under a /test directory

File: internal/desktop_browser.py [hunk 1]
```diff
@@ -359,6 +363,8 @@ def stop(self, job, task):
 
     def wait_for_idle(self, wait_time = 30):
         """Wait for no more than 50% CPU utilization for 400ms"""
+        if self.options.debug:
```
[the remark this answers is not in the document]
Comment (sammeboy635) [reply internal/desktop_browser.py:366]: Very valid point. There is a --noidle command-line flag but it only applies to preventing idle when starting up and not on each test run. Do you think this --noidle flag should prevent all idling?

File: internal/desktop_browser.py [hunk 2]
```diff
@@ -311,6 +314,7 @@ def launch_browser(self, command_line):
 
     def close_browser(self, job, _task):
         """Terminate the browser but don't do all of the cleanup that stop does"""
+        logging.log(8,"Closing the Browser")
```
[the remark this answers is not in the document]
Comment (sammeboy635) [reply internal/desktop_browser.py:317]: I'm not sure if you saw the log that is printed from this line but it is "2022-06-06 15:45:10.395|MainThread|16734|    4|PERFDEB  | Closing the Browser           |desktop_browser.py|close_browser|317". 
The last three fields are | File | Function | Line. 
I can still be more specific if you saw this already.

File: internal/wptutil.py [hunk 3]
```diff
@@ -0,0 +1,363 @@
+from shutil import rmtree
+import os
+import sys
+import logging
+import time
+
```
[the remark this answers is not in the document]
Comment (sammeboy635) [reply internal/wptutil.py:6]: I agree, thanks for that advice!

**Pull request closed without merging**